### PR TITLE
checkout shipping page, close #190

### LINKED ISSUE #190: Checkout Shipping page

### PR DESCRIPTION
**Pull request by schleichardt**
@lauraluiz lets review this together

### CODE REVIEW

**Comment by schleichardt** [comment]
And let's have a look at the failing tests, I think they are maybe hard to maintain.

File: app/inject/SphereSmallCollectionProvider.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.models.Base;
+import io.sphere.sdk.queries.MetaModelQueryDsl;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import java.util.List;
+
+public abstract class SphereSmallCollectionProvider<R, E> extends Base implements Provider<R> {
```
Comment (lauraluiz) [inline app/inject/SphereSmallCollectionProvider.java]: Use internally (or mix with) QueryAll

File: purchase/app/purchase/CheckoutShippingController.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            final Messages messages = messages(userContext);
+            final CheckoutShippingContent content = new CheckoutShippingContent(cart, messages, configuration(), reverseRouter(), userContext, flash(), getCsrfToken(), shippingMethods, productDataConfig);
+            final SunrisePageData pageData = pageData(userContext, content);
+            return ok(templateService().renderToHtml("checkout-shipping", pageData, userContext.locales()));
+        });
+    }
+
+    @AddCSRFToken
+    @RequireCSRFCheck
+    public F.Promise<Result> process(final String languageTag) {
```
Comment (lauraluiz) [inline purchase/app/purchase/CheckoutShippingController.java]: That's some big function... is it possible to split it?

File: purchase/app/purchase/CheckoutShippingFormBean.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                .map(Address::getLastName)
+                .map(lastName -> lastName != null)
+                .orElse(false);
+        setBillingAddressDifferentToBillingAddress(billingAddressDifferentToBillingAddress);
+        setShippingAddress(new AddressFormBean(cart.getShippingAddress(), userContext, messages, configuration));
+        setBillingAddress(new AddressFormBean(cart.getBillingAddress(), userContext, messages, configuration));
+        setShippingMethods(new ShippingMethodsFormBean(cart, shippingMethods));
+    }
+
+    public CheckoutShippingFormBean(final CheckoutShippingFormData checkoutShippingFormData, final ReverseRouter reverseRouter, final String csrfToken, final UserContext userContext, final ShippingMethods shippingMethods, final Messages messages, final Configuration configuration) {
```
Comment (lauraluiz) [inline purchase/app/purchase/CheckoutShippingFormBean.java]: Split it also

File: purchase/app/purchase/SalutationsFieldsBean.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+public class SalutationsFieldsBean {
+    private static final String ALLOWED_TITLES_CONFIG_KEY = "checkout.allowedTitles";
+    private static final String MESSAGE_CONFIG_KEY = "message";
+
+    private List<SelectableData> list;
+
```
Comment (lauraluiz) [inline purchase/app/purchase/SalutationsFieldsBean.java]: Add empty constructor.

**Pull request merged**